io-ballerina/src/io/character/file_xml_io.bal

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+# Read file content as an XML.
+# ```ballerina
+# xml|io:Error content = io:fileReadXml("./resources/myfile.xml");
+# ```
+# + path - XML file path
+# + return - Either a XML or `io:Error`
+public function fileReadXml(@untainted string path) returns @tainted readonly & xml|Error {
+	var fileOpenResult = openReadableFile(path);
+	if (fileOpenResult is ReadableByteChannel) {
+		ReadableCharacterChannel readableCharacterChannel = new (fileOpenResult, DEFAULT_ENCODING);
+        var fileReadResult = readableCharacterChannel.readXml();
+        var closeResult = fileOpenResult.close();
+        if (closeResult is Error) {
+            return closeResult;
+        }
```

Comment:
In this case, however, even though close fails, the file is read isn't it?

Reply:
Yes.

Reply:
In that case, don't we need to return that?

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerina-io/pull/26/commits/0b20a7ab26517624cff62e15f049cacac770151e